docs/migrating_to_v2.rst

```diff
@@ -0,0 +1,255 @@
+| ``await user.avatar_url.read()``                           | ``await user.display_avatar.read()``                                 |
++------------------------------------------------------------+----------------------------------------------------------------------+
+| ``str(emoji.url)``                                         | ``emoji.url``                                                        |
++------------------------------------------------------------+----------------------------------------------------------------------+
+| ``str(emoji.url_as(size=32))``                             | ``emoji.with_size(32).url``                                          |
++------------------------------------------------------------+----------------------------------------------------------------------+
+| ``str(url_as(size=128, static_format="png"))``             | ``emoji.replace(size=32, static_format="png").url``                  |
++------------------------------------------------------------+----------------------------------------------------------------------+
+| ``str(sticker.image_url)``                                 | ``sticker.url``                                                      |
++------------------------------------------------------------+----------------------------------------------------------------------+
+| ``str(partialemoji.url)``                                  | ``partialemoji.url``                                                 |
++------------------------------------------------------------+----------------------------------------------------------------------+
+
+Webhook Changes
+~~~~~~~~~~~~~~~
+
+- ``Webhook`` and ``WebhookMessage`` are now always asynchronouns. For synchronouns use (``requests``), use ``SyncWebhook`` and ``SyncWebhookMessage``.
+- ``WebhookAdapter``, ``AsyncWebhookAdapter``, and ``RequestsWebhookAdapter`` are removed, since they are unnecessary.
+- ``adapter`` arguments of ``Webhook.partial`` and ``Webhook.from_url`` are removed. Sessions are now passed directly to ``partial`` / ``from_url``.
+
+
+.. code-block:: python
+
+    webhook = discord.SyncWebhook.from_url(
+    f"https://discord.com/api/webhooks/{id}/{token}"
+    )
+    webhook.send("Hello from pycord 2.0")
+
+
+.. code-block:: python
+    
+    async with aiohttp.ClientSession() as session:
+    webhook = discord.Webhook.partial(
+        id,
+        token,
+        session=session
+    )
+    await webhook.send("Hello from Pycord 2.0")
+
+
+.. _migrating_2_0_thread_introduced:
+
+Threads Introduced
+------------------
+
+The following methods and attributes can return :class:`Thread` objects:
+
+- :attr:`Message.channel`
+- :meth:`Client.fetch_channel`
+- :meth:`Guild.fetch_channel`
+- :attr:`ChannelNotReadable.argument`
+- :attr:`NSFWChannelRequired.channel`
+- :meth:`Client.get_channel`
+
+.. _migrating_2_0_permission_changes:
+
+Permission Changes
+------------------
+
```

Comment:
Probably add a short sentence on what actually was changed, e.g. 'permissions_in has been removed, permissions_for should be used instead'

Reply:
yeah

Reply:
Like plun said. Describe the what. Not the why x3